csrc/trtllm_flavored_gemm_runner.cu

```diff
@@ -0,0 +1,243 @@
+    TVM_FFI_ICHECK(tactic >= 0 && tactic < gemm.getNumGemmConfigs()) << "Invalid tactic id in run";
+    auto const& config = configs[tactic];
+
+    gemm::gemm::GemmData gemmData = createGemmData(m, n, k);
+
+    // Inputs
+    gemmData.mInputBuffers.mPtrA = b;
+    gemmData.mInputBuffers.mPtrB = a;
+    gemmData.mInputBuffers.mPtrScaleC = cScale;
+
+    // Outputs
+    gemmData.mOutputBuffers.mPtrC = c;
+
+    TVM_FFI_ICHECK(gemm.isValidConfig(config, gemmData))
+        << "The selected tactic points to a TRTLLM-GEN flavored GEMM kernel that is not valid for "
+           "the given problem size.";
+
+    int32_t const multiProcessorCount = [device_index]() {
+      static thread_local int32_t cached_multi_processor_count = -1;
+      static thread_local int cached_device_index = -1;
+
+      if (device_index == cached_device_index && cached_multi_processor_count != -1) {
+        return cached_multi_processor_count;
+      } else {
+        int32_t count;
+        cudaError_t cudaStatus =
+            cudaDeviceGetAttribute(&count, cudaDevAttrMultiProcessorCount, device_index);
+        TVM_FFI_ICHECK(cudaStatus == cudaSuccess)
+            << "Failed to get device attribute: " << cudaGetErrorString(cudaStatus);
+        cached_multi_processor_count = count;
+        cached_device_index = device_index;
+        return count;
+      }
+    }();
+
+    TVM_FFI_ICHECK(gemm.run(config, workspace, gemmData, static_cast<void*>(stream),
+                            multiProcessorCount, true, globalTrtllmGenFlavoredGemmModuleCache) == 0)
+        << "Error occurred when running flavored TRTLLM-GEN GEMM!";
+  }
+
+  std::vector<int64_t> getValidTactics(int64_t m, int64_t n, int64_t k) const {
+    auto const gemm = gemm::gemm::GemmInterface();
+    auto const configs = gemm.getGemmConfigs();
+
+    auto const gemmData = createGemmData(m, n, k);
+
+    std::vector<int64_t> validTactics{};
+    for (auto const& configIndex : mPassingConfigIndices) {
+      auto const& config = configs[configIndex];
+      if (gemm.isValidConfig(config, gemmData)) {
+        validTactics.push_back(configIndex);
+      }
+    }
+    return validTactics;
+  }
+
+ private:
+  TrtllmGenFlavoredGemmRunnerOptions mOptions;
+  std::vector<int64_t> mPassingConfigIndices;
+};
+
+using tvm::ffi::Array;
+using tvm::ffi::Optional;
+
+void trtllm_flavored_gemm(Tensor workspace_buffer, Tensor a, Tensor b, Tensor globalScale,
+                          Tensor out, int64_t tactic) {
+  if (tactic == -1) {
+    TVM_FFI_LOG_AND_THROW(RuntimeError)
+        << "Default tactic is not supported for flavored TRTLLM-GEN GEMM. Please use auto-tuning "
+           "to find the best kernel for your problem.";
+  }
+  CHECK_DEVICE(a, b);
+  CHECK_DEVICE(a, out);
+  CHECK_INPUT(a);
+  CHECK_INPUT(b);
+  CHECK_INPUT(out);
+  CHECK_INPUT(workspace_buffer);
+  CHECK_DIM(2, a);
+  TVM_FFI_ICHECK(b->ndim == 3) << "b must be a block layout matrix (3D tensor with "
```

Comment:
out of curious: can we use `TORCH_CHECK` here to consistent with other code? then maybe no need the tvm header dependency